### ISSUE TITLE
The request option "include" does not work in Parse.User.me()

### ISSUE THREAD

**Issue by frouo** [parse-community/Parse-SDK-JS#1657]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse JS SDK](https://github.com/parse-community/Parse-SDK-JS/releases).

### Issue Description

`await Parse.User.me(sessionToken, { include: "books" })` does not work. 

The "include" key is actually ignored, `books` remains unfetched.

### Steps to reproduce

Create a `Parse.User` with a column of type Pointer array, let's call it "books", add books.

Call `const me = await Parse.User.me(sessionToken, { include: "books" })`

### Actual Outcome

`me.get("books")` returns an array of unfetched `Book`.

### Expected Outcome

Would expect that book data is available.

###  Environment

Server
- Parse Server version: `5.4.0`
- Operating system: `macOS`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `local`

Database
- System (MongoDB or Postgres): `mongodb`
- Database version: `5.0.13`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `local`

Client
- Parse JS SDK version: `3.4.4`

### Fix idea:

https://github.com/parse-community/Parse-SDK-JS/blob/ad429fc7883a9e3fbfe33c1aa70ce7efa7029d77/src/ParseUser.js#L700

```
  static me(sessionToken: string, options?: RequestOptions = {}) {
    const controller = CoreManager.getUserController();
    const meOptions: RequestOptions = {
      sessionToken: sessionToken,
    };
    if (options.useMasterKey) {
      meOptions.useMasterKey = options.useMasterKey;
    }
// FIX
    if (options.include) {
      meOptions.include = options.include;
    }
// end FIX
    const user = new this();
    return controller.me(user, meOptions);
  }
```

**Comment by rpfeil** [comment]
I am absolutely in tears! I have spent weeks trying to figure out why my 'include' functionality stopped working after I switched from REST over to this API. The documentation says nothing about this. Now, after all that suffering, I finally find this. [frouo](https://github.com/frouo), you are a genius! You must have spent so much time tracking down all those details. Fantastic job :-). He took a ton of responsibility to move this ball to the 1 yard line. Then, crickets. Asking him to do more is just plain rude. 3 lines! Isn't there a person who cares about this repo? Is this dying code?

I'm sorry. Thank you for reading my rant. The question is now, will I read through all the contributor guidelines, and spend the time it would take to get this fix in? I built my own system client side to replace these three lines of code. Who has that kind of time, especially after all the time lost.

**Comment by mtrezza** [comment]
Thank you @rpfeil for confirming this issue still exists. It seems it just didn't receive much attention. Does anyone want to open a PR with a failing test case, so we can fix this?

**Comment by rpfeil** [comment]
I see a test that the 'include' query is created. Looking now for a full round trip data test.

https://github.com/parse-community/Parse-SDK-JS/blob/ae875311aac527fc76dc0473ec0c8849dce8e776/src/__tests__/ParseQuery-test.js#L997

**Comment by rpfeil** [comment]
Hmmm. This test seems to be the one that should catch this...
https://github.com/parse-community/Parse-SDK-JS/blob/ae875311aac527fc76dc0473ec0c8849dce8e776/integration/test/ParseQueryTest.js#L1428

**Comment by mtrezza** [comment]
You could open a PR, duplicate and modify one of these tests so that it executes the commands from the issue description with the `me` endpoint. Then we'll see whether it's replicable.

**Comment by rpfeil** [comment]
Just to test this test I'd need to figure out how to run these tests. I've read through a ton of readmes and other info in the code base, found no clue. But also, the integration tests won't run without a local Parse Server and some kind of MongoDb. And then, that is not a fix. If I do all this work, it has to be fixed. 

Completely changing data stack won't take much more time than trying to fix this one. That way, I completely avoid a poorly maintained set of code. How do I justify to my boss that he's paying me to write someone else's code? It strongly feels like I am the Tool, not Parse. 

I could open a PR. But I have bled so much for Parse. I already contribute to other open source projects. I'm cutting my losses.

I tried to help, but can only give so much.

Rick

**Comment by mtrezza** [comment]
There's a guide for first time contributors, see https://github.com/parse-community/Parse-SDK-JS/issues/1657#issuecomment-1396691543. This is all integrated, it automatically launches a Parse Server and MongoDB instance when you run the test.

**Comment by rpfeil** [comment]
You are not listening at all. Please, if you don't have time to read my entire message, then ignore it. Everyone with Parse can ignore it. It's just me venting. Getting out my frustration. I have no interest in continuing to beat my head against this code.

BTW: I was curious, so I took a look at the starters guide. I tried the first few steps. Right away it complains about 'cross-env'. No mention of this as a pre-requisite. Can't imagine how many of these I'd run across if I continued. One is one too many at this point.

**Comment by rpfeil** [comment]
Well, now I'm F'd. I CAN image how many issues I'd run across. I can't stop climbing down this rabbit hole...

1. Next, it talks about a 'spec' directory. No such thing.
2. Then, it talks about a bunch of files with '.spec.' in the middle. No such thing.
3. Then it mentions 'CloudCode.spec.js'. No test file starts with Cloud anything.
4. Oh, now it's talking about a test for 'afterFind'. Which has been commented out.
5. Ah, ok, let's see. the tests are being called fit functions. There are NONE in the code. All converted to it functions.
6. Alright, now I'm supposed to run 'npm test spec/CloudCode.spec.js'. Think this will work? Of course not. But, there is no error message. Someone could spend hours looking for the output file.
7. Now it says, 'The Parse Core team and other contributors are always happy to give tips and pointers and would rather your time not go wasted'. Um, wayyyy too late. If you don't want to waste our time, then don't point us to documentation that hasn't been updated in YEARS!!!!
8. Of course. 'Triggers.js' is mentioned. No test has anything close to triggers in it's name.
9. I'm too tired and sad to continue this. I will only imagine what comes next.

**Comment by rpfeil** [comment]
Please [mtrezza](https://github.com/mtrezza). Never invite another person to contribute to this code base. At least not until you spend a huge amount of time updating your documentation. I also suggest you pull down any language around contributing until you find a way to maintain your documentation.

**Comment by dplewis** [comment]
@mtrezza looks like the contribution guide in the issue bot references the Parse Server instead of the JS SDK

![Image](https://github.com/user-attachments/assets/4b5bc2af-40f9-4930-857a-8246047caac1)

**Comment by rpfeil** [comment]
I... can... not... believe... I... allowed... you... people... to... waste... soooooooooo... much... of... my... time.

Lesson learn. Parse and B4A are dead to me.

Still crying,
Rick

**Comment by rpfeil** [comment]
What happened to the response? Somebody showed that it was the wrong document.

**Comment by dplewis** [comment]
@rpfeil For future reference the contribution guide can be found at the bottom of the ReadMe https://github.com/parse-community/Parse-SDK-JS?tab=readme-ov-file#contributing

I'll do a PR for this issue. I realized there could be refactoring with request options as well as a potential security issue.

For a workaround try
```
const user = await Parse.User.me(sessionToken);
await user.fetchWithInclude(...)
```

Please follow the Parse Community's [Code of Conduct](https://github.com/parse-community/Parse-SDK-JS/blob/alpha/CODE_OF_CONDUCT.md). We understand your frustration but harassment and spam will not be tolerated in the future.

**Comment by rpfeil** [comment]
Please then stop harassing me.

**Comment by dplewis** [comment]
Closing as this is a server issue, duplicate https://github.com/parse-community/parse-server/issues/8095
